tests/logging_helper_test.py

```diff
@@ -4,160 +4,131 @@
-        context):
-    # Tests that client spans collected in a ZipkinLoggingContext get properly
-    # logged.
-    _assert_logged_client_spans(context, None, 'thrift_endpoint')
-
-
-@mock.patch('pyramid_zipkin.logging_helper.copy_endpoint_with_new_service_name',
-            autospec=True)
-def test_zipkin_logging_context_logs_client_span_with_new_endpoint_name(
-        copy_endpoint, context):
-    # Tests that client spans collected in a ZipkinLoggingContext get properly
-    # logged with new "service_name" attribute.
-    copy_endpoint.return_value = 'new_thrift_endpoint'
-    _assert_logged_client_spans(context, 'blargh', 'new_thrift_endpoint')
+    tl_mock.requests = []
+    # Ignore the actual logging part
+    with mock.patch.object(context, 'log_spans'):
+        with context:
+            mock_logger.addHandler.assert_called_once_with(context.handler)
+            assert context.start_timestamp == 42
+            assert tl_mock.requests[0] == context.zipkin_attrs
+        # Make sure the handler and the zipkin attrs are gone
+        mock_logger.removeHandler.assert_called_once_with(context.handler)
+        assert tl_mock.requests == []
+        assert context.log_spans.call_count == 1
 
 
+@mock.patch('pyramid_zipkin.logging_helper.time.time', autospec=True)
+@mock.patch('pyramid_zipkin.logging_helper.log_span', autospec=True)
 @mock.patch('pyramid_zipkin.logging_helper.annotation_list_builder',
             autospec=True)
 @mock.patch('pyramid_zipkin.logging_helper.binary_annotation_list_builder',
             autospec=True)
-@mock.patch('pyramid_zipkin.logging_helper.create_span', autospec=True)
-@mock.patch('pyramid_zipkin.logging_helper.thrift_obj_in_bytes', autospec=True)
-@mock.patch('pyramid_zipkin.logging_helper.log_service_span', autospec=True)
-def _assert_logged_client_spans(
-        logging_context, service_name, thrift_endpoint_mock,
-        log_span, thrift_obj, create_sp, binary_ann, ann):
-    # Sets a bunch of state on the ZipkinLoggingContext, creates some captured
-    # client spans, and asserts that these spans were properly logged.
-    logging_context.start_timestamp = 24
-    logging_context.response_status_code = 200
-    logging_context.registry_settings = {
-            'zipkin.transport_handler': (lambda x, y: None)}
-    spans = [{
-        'annotations': 'ann1', 'binary_annotations': 'bann1',
-        'span_name': 'foo', 'is_client': False,
-        'service_name': service_name,
+@mock.patch('pyramid_zipkin.logging_helper.copy_endpoint_with_new_service_name',
+            autospec=True)
+def test_zipkin_logging_context_log_spans(
+    copy_endpoint_mock, bin_ann_list_builder, ann_list_builder,
+    log_span_mock, time_mock
+):
+    # This lengthy function tests that the logging context properly
+    # logs both client and server spans, while attaching extra annotations
+    # logged throughout the context of the trace.
+    trace_id = '0xf'
+    parent_span_id = '0x1'
```

Comment:
it's annoying, but can we use 16-char hex strings as test data here? it makes it more clear what kind of ids are expected